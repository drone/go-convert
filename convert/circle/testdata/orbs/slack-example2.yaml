version: '2.1'
orbs:
  slack: circleci/slack@4.1
jobs:
  deploy:
    docker:
      - image: cimg/base:stable
    steps:
      - run: echo "deploy my app"
      - slack/notify:
          event: fail
          mentions: '@EngineeringTeam'
          template: basic_fail_1
      - slack/notify:
          event: pass
          template: success_tagged_deploy_1
  test:
    docker:
      - image: cimg/base:stable
    steps:
      - run: echo "test my app"
workflows:
  test-and-deploy:
    jobs:
      - test
      - slack/on-hold:
          context: slack-secrets
          filters:
            tags:
              only: /^v.*/
          requires:
            - test
      - pause_workflow:
          filters:
            tags:
              only: /^v.*/
          requires:
            - test
            - slack/on-hold
          type: approval
      - deploy:
          filters:
            tags:
              only: /^v.*/
          requires:
            - pause_workflow