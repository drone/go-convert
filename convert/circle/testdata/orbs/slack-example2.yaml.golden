kind: pipeline
spec:
  stages:
  - name: test
    spec:
      platform:
        arch: amd64
        os: linux
      runtime:
        spec: {}
        type: cloud
      steps:
      - spec:
          image: cimg/base:stable
          run: |-
            set -eo pipefail
            echo "test my app"
          shell: bash
        type: script
    type: ci
  - name: slack/on-hold
    spec:
      platform:
        arch: amd64
        os: linux
      runtime:
        spec: {}
        type: cloud
      steps:
      - name: circleci/slack
        spec:
          run: echo unable to convert orb circleci/slack/on-hold
        type: script
    type: ci
  - name: deploy
    spec:
      platform:
        arch: amd64
        os: linux
      runtime:
        spec: {}
        type: cloud
      steps:
      - spec:
          image: cimg/base:stable
          run: |-
            set -eo pipefail
            echo "deploy my app"
          shell: bash
        type: script
      - name: notify_slack
        spec:
          image: plugins/slack
          with:
            recipient: '@EngineeringTeam'
            template: basic_fail_1
        type: plugin
      - name: notify_slack
        spec:
          image: plugins/slack
          with:
            template: success_tagged_deploy_1
        type: plugin
    type: ci
version: 1
